### PR TITLE
AP_Scripting: prevent use of closed network socket

### PR DESCRIPTION
**Pull request by tridge**
this gives a lua exception if you try to do this:
```
 sock:recv(1024)
 sock:close()
 sock:recv(1024)
```
without this we dereference null, causing a crash
initially noticed by @magicrub

### CODE REVIEW

File: libraries/AP_Scripting/applets/net_webserver.lua [hunk 1]
```diff
@@ -913,7 +913,10 @@ local function Client(_sock, _idx)
 
    function self.remove()
       DEBUG(string.format("%u: removing client OFFSET=%u", idx, offset))
-      sock:close()
+      if sock then
+         sock:close()
+         sock = nil
```
[the remark this answers is not in the document]
Comment (magicrub) [reply libraries/AP_Scripting/applets/net_webserver.lua:918]: Nope. I was closing and it was causing problems. Adding the =nil after the close solved it

**Comment by magicrub** [comment]
With this change, do we still need to set it to nil?

**Comment by tridge** [comment]
> With this change, do we still need to set it to nil?

yes. This PR means you will get a scripting exception that will stop the script if you use a socket after closing it. Previously it could cause the board to reboot

File: libraries/AP_Scripting/applets/net_webserver.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_Scripting/applets/net_webserver.lua:918]: still needed as the lua variable in the script remains non-nil, it's just that future fetches of the user data in the C++ bindings will fault

**Comment by IamPete1** [comment]
I had a quick look if we could nul the object, but it seems not. https://stackoverflow.com/questions/49582545/set-lua-obj-to-nil-from-c

**Review by IamPete1 — APPROVED** [review]

**Pull request merged**